### PR TITLE
Change bitfields to use variable length Bytes.

### LINKED ISSUE #398: Bitfields should be variable length instead of fixed size.
### Description
As an implementor, I want bitfields to be variable length (using dynamic sizing in SSZ with length prepended) so that Artemis is compliant with the Beacon Chain spec. 

### Acceptance Criteria
* All uses of bitfields through the codebase should use a variable length Bytes value (serialized in using dynamic sizing in SSZ with length prepended) instead of Bytes32.

### Additional Information
This should be expected to have a number of impacts throughout the codebase.

### PR DESCRIPTION
**Pull request by schroedingerscode**
## PR Description
Changes all spec bitfields to be variable length bytes instead of fixes bytes32.

Tasks:
- [x] Change all occurrences of bitfield to variable length Bytes, and update corresponding logic.
- [x] Add tests to ensure variable length bitfields work as intended.

## Fixed Issue(s)
Fixes #398

### CODE REVIEW

File: ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/AttestationTest.java [hunk 1]
```diff
@@ -97,4 +97,37 @@ void rountripSSZ() {
     Bytes sszAttestationBytes = attestation.toBytes();
     assertEquals(attestation, Attestation.fromBytes(sszAttestationBytes));
   }
+
+  @Test
+  void rountripSSZVariableLengthBitfield() {
```
Comment (akhila-raju) [inline ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/AttestationTest.java]: is this supposed to say 'roundtrip'?

Comment (schroedingerscode) [reply ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/AttestationTest.java]: 🤦‍♂️ Yes.

File: ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/SlashableAttestationTest.java [hunk 2]
```diff
@@ -112,4 +110,36 @@ void rountripSSZ() {
     Bytes sszSlashableVoteDataBytes = slashableAttestation.toBytes();
     assertEquals(slashableAttestation, SlashableAttestation.fromBytes(sszSlashableVoteDataBytes));
   }
+
+  @Test
+  void rountripSSZVariableLengthBitfield() {
```
Comment (akhila-raju) [inline ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/SlashableAttestationTest.java]: Also here is this supposed to be 'roundtrip'?

File: ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/state/PendingAttestationTest.java [hunk 3]
```diff
@@ -95,4 +95,45 @@ void rountripSSZ() {
     Bytes sszPendingAttestationBytes = pendingAttestation.toBytes();
     assertEquals(pendingAttestation, PendingAttestation.fromBytes(sszPendingAttestationBytes));
   }
+
+  @Test
+  void rountripSSZVariableLengthBitfield() {
```
Comment (akhila-raju) [inline ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/state/PendingAttestationTest.java]: and here 😛

**Review by akhila-raju — APPROVED** [review]
LGTM although one comment about the test names

**Comment by schroedingerscode** [comment]
Fixed all occurrences. That's probably one of the more copied/pasted typos I've seen in a while.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java [hunk 4]
```diff
@@ -427,13 +427,15 @@ static boolean verify_randao(
    */
   private static boolean verify_bitfields_and_aggregate_signature(
```
Comment (schroedingerscode) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java:428]: @jrhea I wouldn't mind a double check of my logic here. See the first 2 lines in the codeblock here: https://github.com/ethereum/eth2.0-specs/blob/0.4.0/specs/core/0_beacon-chain.md#attestations-1

Comment (jrhea) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java:428]: ya looks right dude

**Review by jrhea — APPROVED** [review]
lgtm

**Pull request merged**